lib/parsers/m3u_parser.rb

```diff
@@ -0,0 +1,22 @@
+class FormatParser::M3UParser
+  include FormatParser::IOUtils
+
+  HEADER = '#EXTM3U'
+
+  def likely_match?(filename)
+    filename =~ /\.(m3u|m3u8)$/i
+  end
+
+  def call(io)
+    io = FormatParser::IOConstraint.new(io)
+
+    header = safe_read(io, 7)
+    return unless HEADER.eql?(header)
+
+    FormatParser::Text.new(
+      format: :m3u,
+      size: io.size
+    )
+  end
+  FormatParser.register_parser new, natures: :text, formats: [:m3u, :m3u8]
```

Comment:
Would `:m3u8` ever be returned as a format by format_parser? Looking at jpg/tif, I'd say not 🤔 

It would be interesting to indeed separate m3u from m3u8, but I don't see it as vital.

Reply:
You are right. It will always return m3u. So m3u8 files will also be recognised as m3u. Actually it might be challenging to distinguish them. I have seen multiple sample files with m3u8 format but they don't have the header to denote the encoding

Reply:
Indeed, the header is rarely used (from what I've seen), and detecting encodings might be quite a hassle. I'd say this is good for now, as you've implemented it.